sdk-http-vertx/src/main/java/dev/restate/sdk/http/vertx/HttpRequestFlowAdapter.java

```diff
@@ -25,7 +26,10 @@ class HttpRequestFlowAdapter implements InvocationFlow.InvocationInputPublisher
   private Flow.Subscriber<? super InvocationFlow.InvocationInput> inputMessagesSubscriber;
   private long subscriberRequest = 0;
 
-  HttpRequestFlowAdapter(HttpServerRequest httpServerRequest) {
+  HttpRequestFlowAdapter(
+      HttpServerRequest httpServerRequest, Protocol.ServiceProtocolVersion serviceProtocolVersion) {
+    assert serviceProtocolVersion == Protocol.ServiceProtocolVersion.V1
+        : "Unsupported service protocol version";
```

Comment:
Not sure why we need this assert here. I would remove it.

Reply:
The idea is to make bumping the service protocol version a conscious decision which might need to touch the encoder/decoder as well.

Reply:
I've changed it into a `IllegalArgumentException`.

Reply:
Not sure this helps, when you bump the protocol you need to do it in the endpoint already, this just adds another point to touch when bumping. Also this class it's a package private class anyway, and only instantiated in the endpoint class.

Reply:
Also this piece of code most likely won't change with a protocol update, but only the internal encoders/decoders. If it will, it's a big enough change that we won't forget to update the request handlers. On the other side, we will most likely forget updating these assertions here (for lambda even worse, we don't have automated tests that check the lambda sdks work fine...)

Reply:
This would be the place where one passes the `serviceProtocolVersion` to the encoder/decoders, I guess. Atm they only support V1 so I didn't bother to pass it through. If you like, then I can also do this.

Reply:
I've removed this check from the code.